Yafc.Model/Model/ProductionTable.cs

```diff
@@ -141,13 +141,17 @@ public bool Search(SearchQuery query) {
         /// <param name="ingredientVariantComparer">The comparer to use when deciding which fluid variants to use.</param>
         /// <param name="selectedFuel">If not <see langword="null"/>, this method will select a crafter or lab that can use this fuel, assuming such an entity exists.
         /// For example, if the selected fuel is coal, the recipe will be configured with a burner assembler/lab if any are available.</param>
-        public void AddRecipe(RecipeOrTechnology recipe, IComparer<Goods> ingredientVariantComparer, Goods? selectedFuel = null) {
+        public void AddRecipe(RecipeOrTechnology recipe, IComparer<Goods> ingredientVariantComparer, Goods? selectedFuel = null, Goods? spentFuel = null) {
             RecipeRow recipeRow = new RecipeRow(this, recipe);
             this.RecordUndo().recipes.Add(recipeRow);
             EntityCrafter? selectedFuelCrafter = selectedFuel?.fuelFor.OfType<EntityCrafter>().Where(e => e.recipes.Contains(recipe)).AutoSelect(DataUtils.FavoriteCrafter);
-            recipeRow.entity = selectedFuelCrafter ?? recipe.crafters.AutoSelect(DataUtils.FavoriteCrafter);
+            EntityCrafter? spentFuelRecipeCrafter = spentFuel is not null ? recipe.crafters.Where(c => c.energy.fuels.OfType<Item>().Any(e => e.fuelResult == spentFuel)).AutoSelect(DataUtils.FavoriteCrafter) : null;
```

Comment:
I suggest that we move away from using lengthy one-liners because they are hard to read.
In this particular case, I suggest to at least put the branch after `?` as a local function, and then expand it further if it's not readable.

Reply:
Done

Reply:
Ideally if it's possible, I suggest to put the whole part that is related to the `spentFuel` input as a private function to make the parent function more readable. Something like `LoadProjectFromPath` in https://github.com/shpaass/yafc-ce/pull/244/commits/dfe72db1e81721baa92912d82a2985e65a96a528